Add fio stress test case with long time test

lubaihua33 opened the pull request:
1. When creating RAID, make the chunk size is 256K.
2. Xfs file system with default settings created on 'md0'.
3. Higher iodepth(256,512,1024,2048)
4. More block sizes(4k,16K,64K,256K) 
5. Add iodepth_batch parameter
6. Longer testing time, about 10 days.

iamshital commented:
**Example 2** 

```
PS C:\GitHub\LISAv2> .\Run-LisaV2.ps1 -TestPlatform Azure -TestLocation centralus -RGIdentifier SS -ARMImageName "Canonical UbuntuServer 16.04-LTS latest" -XMLSecretFile "C:\GitHub\LISAv2\XML\AzureSecrets.xml" -TestCategory BVT -ExcludeTests "BVT-VERIFY-DEPLOYMENT-PROVISION,BVT-VERIFY-DEPLOYMENT-PROVISION-SRIOV"
02/13/2019 16:57:02 : [INFO ] Autogenerated test ID: YL27
02/13/2019 16:57:02 : [INFO ] Found branch information and updated with  excludetests
02/13/2019 16:57:02 : [INFO ] Logging directory: C:\GitHub\LISAv2\TestResults\2019-13-02-08-57-02-4490
02/13/2019 16:57:02 : [INFO ] Setting parameter: TestPlatform = Azure
02/13/2019 16:57:02 : [INFO ] Setting parameter: TestLocation = centralus
02/13/2019 16:57:02 : [INFO ] Setting parameter: ARMImageName = Canonical UbuntuServer 16.04-LTS latest
02/13/2019 16:57:02 : [INFO ] Setting parameter: RGIdentifier = SS
02/13/2019 16:57:02 : [INFO ] Setting parameter: TestCategory = BVT
02/13/2019 16:57:02 : [INFO ] Setting parameter: ExcludeTests = BVT-VERIFY-DEPLOYMENT-PROVISION,BVT-VERIFY-DEPLOYMENT-PROVISION-SRIOV
02/13/2019 16:57:02 : [INFO ] Setting parameter: TestIterations = 1
02/13/2019 16:57:02 : [INFO ] Setting parameter: XMLSecretFile = C:\GitHub\LISAv2\XML\AzureSecrets.xml
02/13/2019 16:57:02 : [INFO ] Test parameters for Azure have been validated successfully. Continue running the test.
02/13/2019 16:57:02 : [INFO ] Updated Test Case xml files.
02/13/2019 16:57:02 : [INFO ] Using provided secrets file: C:\GitHub\LISAv2\XML\AzureSecrets.xml
02/13/2019 16:57:02 : [INFO ] C:\GitHub\LISAv2\XML\AzureSecrets.xml found.
02/13/2019 16:57:02 : [INFO ] ------------------------------------------------------------------
02/13/2019 16:57:02 : [INFO ] Authenticating Azure PS session..
02/13/2019 16:57:03 : [INFO ] Current Subscription : 2cd20493-xxxx-xxxx-xxxx-ab95b63d82c4.
02/13/2019 16:57:03 : [INFO ] ------------------------------------------------------------------
02/13/2019 16:57:03 : [INFO ] Auto selecting storage account : eosgcentralus as per your test region.
02/13/2019 16:57:03 : [INFO ] Updated C:\GitHub\LISAv2\XML\GlobalConfigurations.xml file.
02/13/2019 16:57:03 : [INFO ] ------------------------------------------------------------------
02/13/2019 16:57:04 : [INFO ] SubscriptionName       : Linux Integration Services Dev & Test
02/13/2019 16:57:04 : [INFO ] SubscriptionId         : 2cd20493-xxxx-xxxx-xxxx-ab95b63d82c4
02/13/2019 16:57:04 : [INFO ] User                   : e4284ff8-xxxx-xxxx-xxxx-46675ee7d752
02/13/2019 16:57:04 : [INFO ] ServiceEndpoint        : https://management.core.windows.net/
02/13/2019 16:57:04 : [INFO ] CurrentStorageAccount  : eosgcentralus
02/13/2019 16:57:04 : [INFO ] ------------------------------------------------------------------
02/13/2019 16:57:04 : [INFO ] Setting global variables
02/13/2019 16:57:04 : [INFO ] Validating XML Files from C:\GitHub\LISAv2 folder recursively...
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-IS-ROOT-PASSWORD-DELETED
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-ROOT-DEVICE-TIMEOUT-CHECK
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-HVMODULES-CHECK
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-SWAP-CHECK
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-MTAB-ENTRY-CHECK
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-LAST-CONSOLE-CHECK
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-VERIFY-BOOT-ERROR-WARNINGS
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-VERIFY-SHUTDOWN
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-VERIFY-RESTART
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-VERIFY-UUID-FSTAB
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-VERIFY-SSHD-CONFIG
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-VERIFY-VHD-PREREQUISITES
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-VERIFY-NO-ERROR-IN-LOGS
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-VERIFY-VERBOSE-ENABLED-LOGS
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-VERIFY-BASH-HISTORY
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-PASSWD-POLICY-TEST
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-VERIFY-KVP-DAEMON-STATUS
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-CORE-TIMESYNC-NTP
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-CORE-VERIFY-LIS-MODULES
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-CORE-VERIFY-LIS-MODULES-VERSION
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-CORE-RELOAD-MODULES-SMP
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-CORE-TIME-CLOCKSOURCE
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-CORE-TIME-CLOCKEVENT-UP
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-CORE-TIME-CLOCKEVENT-SMP
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-NET-IFUP-IFDOWN
02/13/2019 16:57:05 : [INFO ] Excluded Test  : BVT-VERIFY-DEPLOYMENT-PROVISION
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-VERIFY-DEPLOYMENT-PROVISION-EPHEMERAL-MANAGED-DISK
02/13/2019 16:57:05 : [INFO ] Excluded Test  : BVT-VERIFY-DEPLOYMENT-PROVISION-SRIOV
02/13/2019 16:57:05 : [INFO ] Collected Test : BVT-VERIFY-HOSTNAME
02/13/2019 16:57:05 : [INFO ] 27 Test Cases have been collected
```

juhlee-microsoft commented on Libraries/CommonFunctions.psm1:
Should be NULL not any string

juhlee-microsoft commented on TestControllers/TestController.psm1:
does it simply hand over the exclusion list? This case shows the user have to know exact name of test case in each xml file, and put them in the parameter. Is there anything we can make it better like pattern-model? It could filter out with string pattern like '*IO*', *4K"? Let's re-visit user-friendly UI.

iamshital commented on TestControllers/TestController.psm1:
That was good suggestion. Implemented it. See updated logs.

iamshital commented:
Example 3 : -ExcludeTest "Testname" does not match any existing tests
---------------------
```
PS C:\GitHub\LISAv2> .\Run-LisaV2.ps1 -TestPlatform Azure -TestLocation centralus -RGIdentifier SS -ARMImageName "Canonical UbuntuServer 16.04-LTS latest" -XMLSecretFile "C:\GitHub\LISAv2\XML\AzureSecrets.xml" -TestCategory BVT -ExcludeTests "BVT-CORE"
02/14/2019 09:22:01 : [INFO ] Autogenerated test ID: GF49
02/14/2019 09:22:01 : [INFO ] Found branch information and updated with  excludetests
02/14/2019 09:22:01 : [INFO ] Logging directory: C:\GitHub\LISAv2\TestResults\2019-14-02-01-22-01-8122
02/14/2019 09:22:01 : [INFO ] Setting parameter: TestPlatform = Azure
02/14/2019 09:22:01 : [INFO ] Setting parameter: TestLocation = centralus
02/14/2019 09:22:01 : [INFO ] Setting parameter: ARMImageName = Canonical UbuntuServer 16.04-LTS latest
02/14/2019 09:22:01 : [INFO ] Setting parameter: RGIdentifier = SS
02/14/2019 09:22:01 : [INFO ] Setting parameter: TestCategory = BVT
02/14/2019 09:22:01 : [INFO ] Setting parameter: ExcludeTests = BVT-CORE
02/14/2019 09:22:01 : [INFO ] Setting parameter: TestIterations = 1
02/14/2019 09:22:01 : [INFO ] Setting parameter: XMLSecretFile = C:\GitHub\LISAv2\XML\AzureSecrets.xml
02/14/2019 09:22:01 : [INFO ] Test parameters for Azure have been validated successfully. Continue running the test.
02/14/2019 09:22:02 : [INFO ] Updated Test Case xml files.
02/14/2019 09:22:02 : [INFO ] Using provided secrets file: C:\GitHub\LISAv2\XML\AzureSecrets.xml
02/14/2019 09:22:02 : [INFO ] C:\GitHub\LISAv2\XML\AzureSecrets.xml found.
02/14/2019 09:22:02 : [INFO ] ------------------------------------------------------------------
02/14/2019 09:22:02 : [INFO ] Authenticating Azure PS session..
02/14/2019 09:22:04 : [INFO ] Current Subscription : 2cd20493-xxxx-xxxx-xxxx-ab95b63d82c4.
02/14/2019 09:22:04 : [INFO ] ------------------------------------------------------------------
02/14/2019 09:22:04 : [INFO ] Auto selecting storage account : eosgcentralus as per your test region.
02/14/2019 09:22:04 : [INFO ] Updated C:\GitHub\LISAv2\XML\GlobalConfigurations.xml file.
02/14/2019 09:22:04 : [INFO ] ------------------------------------------------------------------
02/14/2019 09:22:05 : [INFO ] SubscriptionName       : Linux Integration Services Dev & Test
02/14/2019 09:22:05 : [INFO ] SubscriptionId         : 2cd20493-xxxx-xxxx-xxxx-ab95b63d82c4
02/14/2019 09:22:05 : [INFO ] User                   : e4284ff8-xxxx-xxxx-xxxx-46675ee7d752
02/14/2019 09:22:05 : [INFO ] ServiceEndpoint        : https://management.core.windows.net/
02/14/2019 09:22:05 : [INFO ] CurrentStorageAccount  : eosgcentralus
02/14/2019 09:22:05 : [INFO ] ------------------------------------------------------------------
02/14/2019 09:22:05 : [INFO ] Setting global variables
02/14/2019 09:22:05 : [INFO ] Validating XML Files from C:\GitHub\LISAv2 folder recursively...
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-IS-ROOT-PASSWORD-DELETED
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-ROOT-DEVICE-TIMEOUT-CHECK
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-HVMODULES-CHECK
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-SWAP-CHECK
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-MTAB-ENTRY-CHECK
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-LAST-CONSOLE-CHECK
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-VERIFY-BOOT-ERROR-WARNINGS
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-VERIFY-SHUTDOWN
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-VERIFY-RESTART
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-VERIFY-UUID-FSTAB
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-VERIFY-SSHD-CONFIG
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-VERIFY-VHD-PREREQUISITES
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-VERIFY-NO-ERROR-IN-LOGS
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-VERIFY-VERBOSE-ENABLED-LOGS
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-VERIFY-BASH-HISTORY
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-PASSWD-POLICY-TEST
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-VERIFY-KVP-DAEMON-STATUS
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-CORE-TIMESYNC-NTP
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-CORE-VERIFY-LIS-MODULES
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-CORE-VERIFY-LIS-MODULES-VERSION
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-CORE-RELOAD-MODULES-SMP
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-CORE-TIME-CLOCKSOURCE
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-CORE-TIME-CLOCKEVENT-UP
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-CORE-TIME-CLOCKEVENT-SMP
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-NET-IFUP-IFDOWN
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-VERIFY-DEPLOYMENT-PROVISION
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-VERIFY-DEPLOYMENT-PROVISION-EPHEMERAL-MANAGED-DISK
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-VERIFY-DEPLOYMENT-PROVISION-SRIOV
02/14/2019 09:22:06 : [INFO ] Collected Test : BVT-VERIFY-HOSTNAME
02/14/2019 09:22:06 : [INFO ] 0 Test Cases have been excluded
02/14/2019 09:22:06 : [INFO ] 29 Test Cases have been collected
02/14/2019 09:22:06 : [INFO ] Starting the test
```
Example 4 : -ExcludeTest "TestName-*" (Wildcards)
---------------------
```
PS C:\GitHub\LISAv2> .\Run-LisaV2.ps1 -TestPlatform Azure -TestLocation centralus -RGIdentifier SS -ARMImageName "Canonical UbuntuServer 16.04-LTS latest" -XMLSecretFile "C:\GitHub\LISAv2\XML\AzureSecrets.xml" -TestCategory BVT -ExcludeTests "BVT-CORE-*"
02/14/2019 09:22:14 : [INFO ] Autogenerated test ID: JM87
02/14/2019 09:22:14 : [INFO ] Found branch information and updated with  excludetests
02/14/2019 09:22:14 : [INFO ] Logging directory: C:\GitHub\LISAv2\TestResults\2019-14-02-01-22-14-3857
02/14/2019 09:22:14 : [INFO ] Setting parameter: TestPlatform = Azure
02/14/2019 09:22:14 : [INFO ] Setting parameter: TestLocation = centralus
02/14/2019 09:22:14 : [INFO ] Setting parameter: ARMImageName = Canonical UbuntuServer 16.04-LTS latest
02/14/2019 09:22:14 : [INFO ] Setting parameter: RGIdentifier = SS
02/14/2019 09:22:14 : [INFO ] Setting parameter: TestCategory = BVT
02/14/2019 09:22:14 : [INFO ] Setting parameter: ExcludeTests = BVT-CORE-*
02/14/2019 09:22:14 : [INFO ] Setting parameter: TestIterations = 1
02/14/2019 09:22:14 : [INFO ] Setting parameter: XMLSecretFile = C:\GitHub\LISAv2\XML\AzureSecrets.xml
02/14/2019 09:22:14 : [INFO ] Test parameters for Azure have been validated successfully. Continue running the test.
02/14/2019 09:22:14 : [INFO ] Updated Test Case xml files.
02/14/2019 09:22:14 : [INFO ] Using provided secrets file: C:\GitHub\LISAv2\XML\AzureSecrets.xml
02/14/2019 09:22:14 : [INFO ] C:\GitHub\LISAv2\XML\AzureSecrets.xml found.
02/14/2019 09:22:14 : [INFO ] ------------------------------------------------------------------
02/14/2019 09:22:14 : [INFO ] Authenticating Azure PS session..
02/14/2019 09:22:15 : [INFO ] Current Subscription : 2cd20493-xxxx-xxxx-xxxx-ab95b63d82c4.
02/14/2019 09:22:15 : [INFO ] ------------------------------------------------------------------
02/14/2019 09:22:15 : [INFO ] Auto selecting storage account : eosgcentralus as per your test region.
02/14/2019 09:22:15 : [INFO ] Updated C:\GitHub\LISAv2\XML\GlobalConfigurations.xml file.
02/14/2019 09:22:15 : [INFO ] ------------------------------------------------------------------
02/14/2019 09:22:16 : [INFO ] SubscriptionName       : Linux Integration Services Dev & Test
02/14/2019 09:22:16 : [INFO ] SubscriptionId         : 2cd20493-xxxx-xxxx-xxxx-ab95b63d82c4
02/14/2019 09:22:16 : [INFO ] User                   : e4284ff8-xxxx-xxxx-xxxx-46675ee7d752
02/14/2019 09:22:16 : [INFO ] ServiceEndpoint        : https://management.core.windows.net/
02/14/2019 09:22:16 : [INFO ] CurrentStorageAccount  : eosgcentralus
02/14/2019 09:22:16 : [INFO ] ------------------------------------------------------------------
02/14/2019 09:22:16 : [INFO ] Setting global variables
02/14/2019 09:22:16 : [INFO ] Validating XML Files from C:\GitHub\LISAv2 folder recursively...
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-IS-ROOT-PASSWORD-DELETED
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-ROOT-DEVICE-TIMEOUT-CHECK
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-HVMODULES-CHECK
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-SWAP-CHECK
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-MTAB-ENTRY-CHECK
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-LAST-CONSOLE-CHECK
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-VERIFY-BOOT-ERROR-WARNINGS
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-VERIFY-SHUTDOWN
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-VERIFY-RESTART
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-VERIFY-UUID-FSTAB
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-VERIFY-SSHD-CONFIG
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-VERIFY-VHD-PREREQUISITES
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-VERIFY-NO-ERROR-IN-LOGS
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-VERIFY-VERBOSE-ENABLED-LOGS
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-VERIFY-BASH-HISTORY
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-PASSWD-POLICY-TEST
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-VERIFY-KVP-DAEMON-STATUS
02/14/2019 09:22:16 : [INFO ] Excluded Test  : BVT-CORE-TIMESYNC-NTP [Wildcards match]
02/14/2019 09:22:16 : [INFO ] Excluded Test  : BVT-CORE-VERIFY-LIS-MODULES [Wildcards match]
02/14/2019 09:22:16 : [INFO ] Excluded Test  : BVT-CORE-VERIFY-LIS-MODULES-VERSION [Wildcards match]
02/14/2019 09:22:16 : [INFO ] Excluded Test  : BVT-CORE-RELOAD-MODULES-SMP [Wildcards match]
02/14/2019 09:22:16 : [INFO ] Excluded Test  : BVT-CORE-TIME-CLOCKSOURCE [Wildcards match]
02/14/2019 09:22:16 : [INFO ] Excluded Test  : BVT-CORE-TIME-CLOCKEVENT-UP [Wildcards match]
02/14/2019 09:22:16 : [INFO ] Excluded Test  : BVT-CORE-TIME-CLOCKEVENT-SMP [Wildcards match]
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-NET-IFUP-IFDOWN
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-VERIFY-DEPLOYMENT-PROVISION
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-VERIFY-DEPLOYMENT-PROVISION-EPHEMERAL-MANAGED-DISK
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-VERIFY-DEPLOYMENT-PROVISION-SRIOV
02/14/2019 09:22:16 : [INFO ] Collected Test : BVT-VERIFY-HOSTNAME
02/14/2019 09:22:16 : [INFO ] 7 Test Cases have been excluded
02/14/2019 09:22:16 : [INFO ] 22 Test Cases have been collected
02/14/2019 09:22:16 : [INFO ] Starting the test
```

Example 5 : -ExcludeTest "TestName1,TestName-*" (Combination of wildcards and exact testnames)
-----------------------
```
PS C:\GitHub\LISAv2> .\Run-LisaV2.ps1 -TestPlatform Azure -TestLocation centralus -RGIdentifier SS -ARMImageName "Canonical UbuntuServer 16.04-LTS latest" -XMLSecretFile "C:\GitHub\LISAv2\XML\AzureSecrets.xml" -TestCategory BVT -ExcludeTests "BVT-CORE-TIME-CLOCKEVENT-SMP,BVT-VERIFY-VERBOSE-ENABLED-LOGS,BVT-VERIFY*"
02/14/2019 09:23:03 : [INFO ] Autogenerated test ID: GE51
02/14/2019 09:23:03 : [INFO ] Found branch information and updated with  excludetests
02/14/2019 09:23:03 : [INFO ] Logging directory: C:\GitHub\LISAv2\TestResults\2019-14-02-01-23-03-8249
02/14/2019 09:23:03 : [INFO ] Setting parameter: TestPlatform = Azure
02/14/2019 09:23:03 : [INFO ] Setting parameter: TestLocation = centralus
02/14/2019 09:23:03 : [INFO ] Setting parameter: ARMImageName = Canonical UbuntuServer 16.04-LTS latest
02/14/2019 09:23:03 : [INFO ] Setting parameter: RGIdentifier = SS
02/14/2019 09:23:03 : [INFO ] Setting parameter: TestCategory = BVT
02/14/2019 09:23:03 : [INFO ] Setting parameter: ExcludeTests = BVT-CORE-TIME-CLOCKEVENT-SMP,BVT-VERIFY-VERBOSE-ENABLED-LOGS,BVT-VERIFY*
02/14/2019 09:23:03 : [INFO ] Setting parameter: TestIterations = 1
02/14/2019 09:23:03 : [INFO ] Setting parameter: XMLSecretFile = C:\GitHub\LISAv2\XML\AzureSecrets.xml
02/14/2019 09:23:03 : [INFO ] Test parameters for Azure have been validated successfully. Continue running the test.
02/14/2019 09:23:03 : [INFO ] Updated Test Case xml files.
02/14/2019 09:23:04 : [INFO ] Using provided secrets file: C:\GitHub\LISAv2\XML\AzureSecrets.xml
02/14/2019 09:23:04 : [INFO ] C:\GitHub\LISAv2\XML\AzureSecrets.xml found.
02/14/2019 09:23:04 : [INFO ] ------------------------------------------------------------------
02/14/2019 09:23:04 : [INFO ] Authenticating Azure PS session..
02/14/2019 09:23:06 : [INFO ] Current Subscription : 2cd20493-xxxx-xxxx-xxxx-ab95b63d82c4.
02/14/2019 09:23:06 : [INFO ] ------------------------------------------------------------------
02/14/2019 09:23:06 : [INFO ] Auto selecting storage account : eosgcentralus as per your test region.
02/14/2019 09:23:06 : [INFO ] Updated C:\GitHub\LISAv2\XML\GlobalConfigurations.xml file.
02/14/2019 09:23:06 : [INFO ] ------------------------------------------------------------------
02/14/2019 09:23:07 : [INFO ] SubscriptionName       : Linux Integration Services Dev & Test
02/14/2019 09:23:07 : [INFO ] SubscriptionId         : 2cd20493-xxxx-xxxx-xxxx-ab95b63d82c4
02/14/2019 09:23:07 : [INFO ] User                   : e4284ff8-xxxx-xxxx-xxxx-46675ee7d752
02/14/2019 09:23:07 : [INFO ] ServiceEndpoint        : https://management.core.windows.net/
02/14/2019 09:23:07 : [INFO ] CurrentStorageAccount  : eosgcentralus
02/14/2019 09:23:07 : [INFO ] ------------------------------------------------------------------
02/14/2019 09:23:07 : [INFO ] Setting global variables
02/14/2019 09:23:07 : [INFO ] Validating XML Files from C:\GitHub\LISAv2 folder recursively...
02/14/2019 09:23:07 : [INFO ] Collected Test : BVT-IS-ROOT-PASSWORD-DELETED
02/14/2019 09:23:07 : [INFO ] Collected Test : BVT-ROOT-DEVICE-TIMEOUT-CHECK
02/14/2019 09:23:07 : [INFO ] Collected Test : BVT-HVMODULES-CHECK
02/14/2019 09:23:07 : [INFO ] Collected Test : BVT-SWAP-CHECK
02/14/2019 09:23:07 : [INFO ] Collected Test : BVT-MTAB-ENTRY-CHECK
02/14/2019 09:23:07 : [INFO ] Collected Test : BVT-LAST-CONSOLE-CHECK
02/14/2019 09:23:07 : [INFO ] Excluded Test  : BVT-VERIFY-BOOT-ERROR-WARNINGS [Wildcards match]
02/14/2019 09:23:07 : [INFO ] Excluded Test  : BVT-VERIFY-SHUTDOWN [Wildcards match]
02/14/2019 09:23:08 : [INFO ] Excluded Test  : BVT-VERIFY-RESTART [Wildcards match]
02/14/2019 09:23:08 : [INFO ] Excluded Test  : BVT-VERIFY-UUID-FSTAB [Wildcards match]
02/14/2019 09:23:08 : [INFO ] Excluded Test  : BVT-VERIFY-SSHD-CONFIG [Wildcards match]
02/14/2019 09:23:08 : [INFO ] Excluded Test  : BVT-VERIFY-VHD-PREREQUISITES [Wildcards match]
02/14/2019 09:23:08 : [INFO ] Excluded Test  : BVT-VERIFY-NO-ERROR-IN-LOGS [Wildcards match]
02/14/2019 09:23:08 : [INFO ] Excluded Test  : BVT-VERIFY-VERBOSE-ENABLED-LOGS [Exact match]
02/14/2019 09:23:08 : [INFO ] Excluded Test  : BVT-VERIFY-VERBOSE-ENABLED-LOGS [Wildcards match]
02/14/2019 09:23:08 : [INFO ] Excluded Test  : BVT-VERIFY-BASH-HISTORY [Wildcards match]
02/14/2019 09:23:08 : [INFO ] Collected Test : BVT-PASSWD-POLICY-TEST
02/14/2019 09:23:08 : [INFO ] Excluded Test  : BVT-VERIFY-KVP-DAEMON-STATUS [Wildcards match]
02/14/2019 09:23:08 : [INFO ] Collected Test : BVT-CORE-TIMESYNC-NTP
02/14/2019 09:23:08 : [INFO ] Collected Test : BVT-CORE-VERIFY-LIS-MODULES
02/14/2019 09:23:08 : [INFO ] Collected Test : BVT-CORE-VERIFY-LIS-MODULES-VERSION
02/14/2019 09:23:08 : [INFO ] Collected Test : BVT-CORE-RELOAD-MODULES-SMP
02/14/2019 09:23:08 : [INFO ] Collected Test : BVT-CORE-TIME-CLOCKSOURCE
02/14/2019 09:23:08 : [INFO ] Collected Test : BVT-CORE-TIME-CLOCKEVENT-UP
02/14/2019 09:23:08 : [INFO ] Excluded Test  : BVT-CORE-TIME-CLOCKEVENT-SMP [Exact match]
02/14/2019 09:23:08 : [INFO ] Collected Test : BVT-NET-IFUP-IFDOWN
02/14/2019 09:23:08 : [INFO ] Excluded Test  : BVT-VERIFY-DEPLOYMENT-PROVISION [Wildcards match]
02/14/2019 09:23:08 : [INFO ] Excluded Test  : BVT-VERIFY-DEPLOYMENT-PROVISION-EPHEMERAL-MANAGED-DISK [Wildcards match]
02/14/2019 09:23:08 : [INFO ] Excluded Test  : BVT-VERIFY-DEPLOYMENT-PROVISION-SRIOV [Wildcards match]
02/14/2019 09:23:08 : [INFO ] Excluded Test  : BVT-VERIFY-HOSTNAME [Wildcards match]
02/14/2019 09:23:08 : [INFO ] 15 Test Cases have been excluded
02/14/2019 09:23:08 : [INFO ] 14 Test Cases have been collected
02/14/2019 09:23:08 : [INFO ] Starting the test
```

iamshital commented on Libraries/CommonFunctions.psm1:
If we make it NULL, then we need to add one more if block to verify if $ExcludeTests -ne NULL, and then proceed for actual checking.

Note, at line 166, I'm doing $ExcludeTests.Split(",")
So if the -ExcludeTests is null, then we'll get exception here.

Assigning a specific string, avoids all this hassle.

iamshital commented on TestControllers/TestController.psm1:
By the way, `*4K` is wrong pattern. Powershell results in error : `parsing "*4K" - Quantifier {x,y} following nothing.`
You can use `.*4K` instead.

**Example:**
```
PS C:\GitHub\LISAv2> .\Run-LisaV2.ps1 -TestPlatform Azure -TestLocation centralus -RGIdentifier SS -ARMImageName "Canonical UbuntuServer 16.04-LTS latest" -XMLSecretFile "C:\GitHub\LISAv2\XML\AzureSecrets.xml" -TestArea Storage -ExcludeTests ".*4K.*"
02/14/2019 09:50:26 : [INFO ] Autogenerated test ID: CK34
02/14/2019 09:50:26 : [INFO ] Found branch information and updated with  excludetests
02/14/2019 09:50:26 : [INFO ] Logging directory: C:\GitHub\LISAv2\TestResults\2019-14-02-01-50-26-3135
02/14/2019 09:50:26 : [INFO ] Setting parameter: TestPlatform = Azure
02/14/2019 09:50:26 : [INFO ] Setting parameter: TestLocation = centralus
02/14/2019 09:50:26 : [INFO ] Setting parameter: ARMImageName = Canonical UbuntuServer 16.04-LTS latest
02/14/2019 09:50:26 : [INFO ] Setting parameter: RGIdentifier = SS
02/14/2019 09:50:26 : [INFO ] Setting parameter: TestArea = Storage
02/14/2019 09:50:26 : [INFO ] Setting parameter: ExcludeTests = .*4K.*
02/14/2019 09:50:26 : [INFO ] Setting parameter: TestIterations = 1
02/14/2019 09:50:26 : [INFO ] Setting parameter: XMLSecretFile = C:\GitHub\LISAv2\XML\AzureSecrets.xml
02/14/2019 09:50:26 : [INFO ] Test parameters for Azure have been validated successfully. Continue running the test.
02/14/2019 09:50:26 : [INFO ] Updated Test Case xml files.
02/14/2019 09:50:26 : [INFO ] Using provided secrets file: C:\GitHub\LISAv2\XML\AzureSecrets.xml
02/14/2019 09:50:26 : [INFO ] C:\GitHub\LISAv2\XML\AzureSecrets.xml found.
02/14/2019 09:50:26 : [INFO ] ------------------------------------------------------------------
02/14/2019 09:50:26 : [INFO ] Authenticating Azure PS session..
02/14/2019 09:50:27 : [INFO ] Current Subscription : 2cd20493-xxxx-xxxx-xxxx-ab95b63d82c4.
02/14/2019 09:50:27 : [INFO ] ------------------------------------------------------------------
02/14/2019 09:50:27 : [INFO ] Auto selecting storage account : eosgcentralus as per your test region.
02/14/2019 09:50:27 : [INFO ] Updated C:\GitHub\LISAv2\XML\GlobalConfigurations.xml file.
02/14/2019 09:50:27 : [INFO ] ------------------------------------------------------------------
02/14/2019 09:50:27 : [INFO ] SubscriptionName       : Linux Integration Services Dev & Test
02/14/2019 09:50:27 : [INFO ] SubscriptionId         : 2cd20493-xxxx-xxxx-xxxx-ab95b63d82c4
02/14/2019 09:50:27 : [INFO ] User                   : e4284ff8-xxxx-xxxx-xxxx-46675ee7d752
02/14/2019 09:50:27 : [INFO ] ServiceEndpoint        : https://management.core.windows.net/
02/14/2019 09:50:27 : [INFO ] CurrentStorageAccount  : eosgcentralus
02/14/2019 09:50:27 : [INFO ] ------------------------------------------------------------------
02/14/2019 09:50:27 : [INFO ] Setting global variables
02/14/2019 09:50:28 : [INFO ] Validating XML Files from C:\GitHub\LISAv2 folder recursively...
02/14/2019 09:50:28 : [INFO ] Collected Test : AZURE-WINDOWS-NESTED-HYPERV-STORAGE-SINGLE-DISK
02/14/2019 09:50:28 : [INFO ] Collected Test : AZURE-WINDOWS-NESTED-HYPERV-STORAGE-MULTIDISK
02/14/2019 09:50:28 : [INFO ] Excluded Test  : PERF-STORAGE-4K-IO [Wildcards match]
02/14/2019 09:50:28 : [INFO ] Excluded Test  : PERF-STORAGE-1024K-IO [Wildcards match]
02/14/2019 09:50:28 : [INFO ] Excluded Test  : PERF-STORAGE-OVER-NFS-Synthetic-TCP-4K [Wildcards match]
02/14/2019 09:50:28 : [INFO ] Excluded Test  : PERF-STORAGE-OVER-NFS-Synthetic-UDP-4K [Wildcards match]
02/14/2019 09:50:28 : [INFO ] Excluded Test  : PERF-STORAGE-OVER-NFS-SRIOV-TCP-4K [Wildcards match]
02/14/2019 09:50:28 : [INFO ] Excluded Test  : PERF-STORAGE-OVER-NFS-SRIOV-UDP-4K [Wildcards match]
02/14/2019 09:50:28 : [INFO ] 6 Test Cases have been excluded
02/14/2019 09:50:28 : [INFO ] 2 Test Cases have been collected
02/14/2019 09:50:28 : [INFO ] Starting the test
```

juhlee-microsoft commented on Libraries/CommonFunctions.psm1:
I am not sure that is hassle. Try to design better automation. Putting a certain string value in the condition of no exclusion, it does not make sense in your 'ExcludeTests' variable. That variable should have value for exclusion tests, right?

juhlee-microsoft commented on TestControllers/TestController.psm1:
Can you unblock PowerShell error? Your proposal does not follow common regular expression. Better idea? Using '.' in front of '*' does not sound good. Please propose better design.

iamshital commented on TestControllers/TestController.psm1:
@juhlee-microsoft Fixed.
Note: ".*" is a common practice in regex world. But powershell does not support starting with "*"

iamshital commented on Libraries/CommonFunctions.psm1:
@juhlee-microsoft Fixed. Removed assigned variable and added if condition to check if -ExcludeTests is not empty.

iamshital commented on TestControllers/TestController.psm1:
**Updated Logs:**

```
PS D:\GitHub\LISAv2> .\Run-LisaV2.ps1 -TestPlatform Azure -TestLocation centralus -RGIdentifier SS -ARMImageName "Canonical UbuntuServer 16.04-LTS latest" -XMLSecretFile "D:\GitHub\LISAv2\XML\AzureSecrets.xml" -TestArea Storage -ExcludeTests "*4K"
02-14-2019 18:52:49 : [INFO ] Autogenerated test ID: CV96
02-14-2019 18:52:49 : [INFO ] Found branch information and updated with  excludetests
02-14-2019 18:52:49 : [INFO ] Logging directory: D:\GitHub\LISAv2\TestResults\2019-15-02-00-22-49-4044
02-14-2019 18:52:49 : [INFO ] Setting parameter: TestPlatform = Azure
02-14-2019 18:52:49 : [INFO ] Setting parameter: TestLocation = centralus
02-14-2019 18:52:49 : [INFO ] Setting parameter: ARMImageName = Canonical UbuntuServer 16.04-LTS latest
02-14-2019 18:52:49 : [INFO ] Setting parameter: RGIdentifier = SS
02-14-2019 18:52:49 : [INFO ] Setting parameter: TestArea = Storage
02-14-2019 18:52:49 : [INFO ] Setting parameter: ExcludeTests = *4K
02-14-2019 18:52:49 : [INFO ] Setting parameter: TestIterations = 1
02-14-2019 18:52:49 : [INFO ] Setting parameter: XMLSecretFile = D:\GitHub\LISAv2\XML\AzureSecrets.xml
02-14-2019 18:52:49 : [INFO ] Test parameters for Azure have been validated successfully. Continue running the test.
02-14-2019 18:52:49 : [INFO ] Updated Test Case xml files.
02-14-2019 18:52:49 : [INFO ] Using provided secrets file: D:\GitHub\LISAv2\XML\AzureSecrets.xml
02-14-2019 18:52:49 : [INFO ] D:\GitHub\LISAv2\XML\AzureSecrets.xml found.
02-14-2019 18:52:49 : [INFO ] ------------------------------------------------------------------
02-14-2019 18:52:49 : [INFO ] Authenticating Azure PS session..
02-14-2019 18:52:52 : [INFO ] Current Subscription : 2cd20493-xxxx-xxxx-xxxx-ab95b63d82c4.
02-14-2019 18:52:52 : [INFO ] ------------------------------------------------------------------
02-14-2019 18:52:52 : [INFO ] Auto selecting storage account : eosgcentralus as per your test region.
02-14-2019 18:52:52 : [INFO ] Updated D:\GitHub\LISAv2\XML\GlobalConfigurations.xml file.
02-14-2019 18:52:52 : [INFO ] ------------------------------------------------------------------
02-14-2019 18:52:53 : [INFO ] SubscriptionName       : Linux Integration Services Dev & Test
02-14-2019 18:52:53 : [INFO ] SubscriptionId         : 2cd20493-xxxx-xxxx-xxxx-ab95b63d82c4
02-14-2019 18:52:53 : [INFO ] User                   : e4284ff8-xxxx-xxxx-xxxx-46675ee7d752
02-14-2019 18:52:53 : [INFO ] ServiceEndpoint        : https://management.core.windows.net/
02-14-2019 18:52:53 : [INFO ] CurrentStorageAccount  : eosgcentralus
02-14-2019 18:52:53 : [INFO ] ------------------------------------------------------------------
02-14-2019 18:52:53 : [INFO ] Setting global variables
02-14-2019 18:52:53 : [INFO ] Validating XML Files from D:\GitHub\LISAv2 folder recursively...
02-14-2019 18:52:55 : [INFO ] Collected Test : AZURE-WINDOWS-NESTED-HYPERV-STORAGE-SINGLE-DISK
02-14-2019 18:52:55 : [INFO ] Collected Test : AZURE-WINDOWS-NESTED-HYPERV-STORAGE-MULTIDISK
02-14-2019 18:52:55 : [INFO ] Excluded Test  : PERF-STORAGE-4K-IO [Wildcards match]
02-14-2019 18:52:55 : [INFO ] Excluded Test  : PERF-STORAGE-1024K-IO [Wildcards match]
02-14-2019 18:52:55 : [INFO ] Excluded Test  : PERF-STORAGE-OVER-NFS-Synthetic-TCP-4K [Wildcards match]
02-14-2019 18:52:55 : [INFO ] Excluded Test  : PERF-STORAGE-OVER-NFS-Synthetic-UDP-4K [Wildcards match]
02-14-2019 18:52:55 : [INFO ] Excluded Test  : PERF-STORAGE-OVER-NFS-SRIOV-TCP-4K [Wildcards match]
02-14-2019 18:52:55 : [INFO ] Excluded Test  : PERF-STORAGE-OVER-NFS-SRIOV-UDP-4K [Wildcards match]
02-14-2019 18:52:55 : [INFO ] 6 Test Cases have been excluded
02-14-2019 18:52:55 : [INFO ] 2 Test Cases have been collected
02-14-2019 18:52:55 : [INFO ] Starting the test
```

LiliDeng commented on Libraries/CommonFunctions.psm1:
if ($ExcludeTestMatched) is enough?

iamshital commented on Libraries/CommonFunctions.psm1:
@LiliDeng , Yes, I've verified it.

LiliDeng commented on Libraries/CommonFunctions.psm1:
I mean no need compare it with $ture

iamshital commented on Libraries/CommonFunctions.psm1:
right.. will fix it.

lizzha commented:
@iamshital could you update document for how to use the parameter, maybe add some examples?

iamshital commented:
> @iamshital could you update document for how to use the parameter, maybe add some examples?

Sure.. I'm on it.

juhlee-microsoft commented on Run-LisaV2.ps1:
@iamshital Used 'double quotations' and 'single quotation' here and below in mixed. Is it by-design or no restriction?

iamshital commented on Run-LisaV2.ps1:
@juhlee-microsoft  As long as you are not using powershell special character (e.g. `$`), then there is no restriction.

juhlee-microsoft commented on Run-LisaV2.ps1:
Please keep in double quotations.

iamshital commented on Run-LisaV2.ps1:
@juhlee-microsoft Done, and also verified with double quoted regex. (as shown in example)

LiliDeng committed:
Fix collect_VM_properties

harz566 committed:
enable perf_ntttcp.sh usage for generic scenario (#734)

juhlee-microsoft committed:
Update KDUMP-TestScript.ps1

lubaihua33 committed:
Add somme changes for supporting MSFT kernel

juhlee-microsoft committed:
Update KDUMP-TestScript.ps1

LiliDeng committed:
Fix KDUMP issue found in ubuntu pipeline.

LiliDeng committed:
Fix based on comments.

LiliDeng committed:
Fix issue when install customlis

LiliDeng committed:
Fix based on comments.

LiliDeng committed:
Fix issue hit on Ubuntu 20.04

lubaihua33 committed:
Add 2>&1 to record log

lubaihua33 committed:
Add a new fio stress test with long time test

lubaihua33 committed:
perf_fio: When test is failed, we also need to collect logs

lubaihua33 committed:
perf_fio: Have some changes for fio stress test.

Delete the numJobDefine parameter. When the test result is pass or fail,
we collect the log.
